### ISSUE TITLE
Dash deemed illegal

### ISSUE THREAD

**Issue by tomekwi** [Kuniwak/jsdoctypeparser#15]
I use dashed names for my custom types in jsDoc to distinguish them from identifiers in code. jsDoc hs nothing against it. But jsdoctypeparser does:

```sh
$ cat <<——— | dox
/**
 * @typedef my-type
 */
———

…/dox/node_modules/jsdoctypeparser/lib/index.js:53
          throw new Lexer.SyntaxError(e.message, typeExp, e.offset);
                ^
TypeLexerSyntaxError: Expected "!", "#", ".", ".<", "<", "=", "?", "[]", "|", "~", [ \t\r\n ], [a-zA-Z0-9_$] or end of input but "-" found.:
y-typ
^
```

**Comment by Kuniwak** [comment]
Hmm...

OK, I'm gonna accept `-` in the identifier token on v2.0.

But, I think `-` is not good, because a type name identifier should equal to the JavaScript object/variable name and we can not use `-` as the identifier in JavaScript... 

```javascript
>>> var var-has-hyphen = 'is not accepted';
Uncaught SyntaxError: Unexpected token var
```

**Comment by tomekwi** [comment]
Yeah, I know that – and that’s exactly the reason why I use them :)

> to distinguish them from identifiers in code

**Comment by Kuniwak** [comment]
The feature was released with v2.0.0-alpha.
See [online demo](http://kuniwak.github.io/jsdoctypeparser/).

**Comment by tomekwi** [comment]
:+
